SingularityUI/app/components/common/modalButtons/RedeployModal.jsx

```diff
@@ -0,0 +1,117 @@
+import React, { Component, PropTypes } from 'react';
+import { Link } from 'react-router';
+import { connect } from 'react-redux';
+import { Alert } from 'react-bootstrap';
+import JSONTree from 'react-json-tree';
+import { JSONTreeTheme } from '../../../thirdPartyConfigurations';
+
+import { SaveDeploy } from '../../../actions/api/deploys';
+
+import FormModal from '../modal/FormModal';
+import SimpleModal from '../modal/SimpleModal';
+import Utils from '../../../utils';
+
+class RedeployModal extends Component {
+  static propTypes = {
+    requestId: PropTypes.string.isRequired,
+    saveDeploy: PropTypes.func,
+    doAfterRedeploy: PropTypes.func,
+    deploy: PropTypes.object
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+
+  showSuccessModal() {
+    this.refs.successModal.show();
+  }
+
+  show() {
+    this.refs.redeployModal.show();
+  }
+
+  confirmRedeploy(data) {
+    this.props.saveDeploy(data.deployId).then((response) => {
+      if (response.statusCode === 200) {
+        this.setState({newDeployId: data.deployId});
+        this.showSuccessModal();
```

Comment:
Instead of having a full-fledged modal to show success that the user then has to dismiss, would it make more UX sense to display a `Messenger` notification instead?